bundles/core/org.openhab.core/src/main/java/org/openhab/core/types/UnDefType.java

```diff
@@ -34,4 +34,20 @@ public String format(String pattern) {
 		return String.format(pattern, this.toString());
 	}
 
+	/**
+	 * Returns the enum constant with the specified name.
+	 */
+	public static UnDefType parse(String name) {
+		if (name == null) {
+			throw new NullPointerException("Name is null");
+		}
+		if (UNDEF.toString().equals(name)) {
+			return UNDEF;
+		} else if (NULL.toString().equals(name)) {
+			return NULL;
+		}
+		throw new IllegalArgumentException("No enum constant "
+				+ UnDefType.class.getCanonicalName() + "." + name);
+	}

```

Comment:
Why not `valueOf` instead of `parse` so you don't need a special case in `TypeParser`?  What do you mean in the comment "valueOf does not work because of the toString method"?

Also, what is the likely impact of this core change?